etcdserver/etcdhttp/client.go

```diff
@@ -281,6 +283,35 @@ func (h *statsHandler) serveLeader(w http.ResponseWriter, r *http.Request) {
 	w.Write(stats)
 }
 
+// TODO: change etcdserver to raft interface when we have it.
+//       add test for healthHeadler when we have the interface ready.
+func healthHandler(server *etcdserver.EtcdServer) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if !allowMethod(w, r.Method, "GET") {
+			return
+		}
+
+		if uint64(server.Leader()) == raft.None {
+			http.Error(w, `"{health": "false"}`, http.StatusServiceUnavailable)
+			return
+		}
+
+		// wait for raft's progress
+		index := server.Index()
+		for i := 0; i < 3; i++ {
+			time.Sleep(250 * time.Millisecond)
```

Comment:
Not worth it. We do not care about the sync interval. The general idea is etcd should make a progress within 0.5 second.